src/apis/orderbookapi.py

```diff
@@ -0,0 +1,173 @@
+"""
+OrderbookAPI for fetching relevant data using the CoW Swap Orderbook API.
+"""
+# pylint: disable=logging-fstring-interpolation
+
+from typing import Any, Optional
+from json import loads
+import requests
+from src.helper_functions import get_logger
+from src.models import Trade, OrderExecution
+from src.constants import (
+    header,
+    REQUEST_TIMEOUT,
+    SUCCESS_CODE,
+    FAIL_CODE,
+)
+
+
+class OrderbookAPI:
+    """
+    Class for fetching data from a Web3 API.
+    """
+
+    def __init__(self):
+        self.logger = get_logger()
+
+    def get_solver_competition_data(self, tx_hash: str) -> Optional[dict[str, Any]]:
+        """
+        Get solver competition data from a transaction hash.
+        The returned dict follows the schema outlined here:
+        https://api.cow.fi/docs/#/default/get_api_v1_solver_competition_by_tx_hash__tx_hash_
+        """
+        prod_endpoint_url = (
+            "https://api.cow.fi/mainnet/api/v1/solver_competition"
+            f"/by_tx_hash/{tx_hash}"
+        )
+        barn_endpoint_url = (
+            "https://barn.api.cow.fi/mainnet/api/v1"
+            f"/solver_competition/by_tx_hash/{tx_hash}"
+        )
+        try:
+            json_competition_data = requests.get(
+                prod_endpoint_url,
+                headers=header,
+                timeout=REQUEST_TIMEOUT,
+            )
+            if json_competition_data.status_code == SUCCESS_CODE:
+                solver_competition_data = loads(json_competition_data.text)
```

Comment:
Nit. Although it may seem inconvenient, I think it's common practice to use the full json.loads here which can be archived in the import statement above by simply saying import json

The reason, I believe, is because loads alone doesn't provide enough context as to what is happening but when together with json is perfectly clear.